deploy.sh

```
@@ -16,35 +16,137 @@ check_docker_daemon() {
+
+# Function to download and run Docker Compose
 run_docker_compose() {
   local compose_url=$1
-  local compose_file=$(basename $compose_url)
-  
-  echo "Downloading $compose_file from $compose_url..."
-  curl -O $compose_url  # Download the file
-  
-  echo "Running docker-compose with $compose_file..."
-  docker-compose -f $compose_file up -d  # Run docker-compose with the downloaded file
+  local compose_file=$(basename "$compose_url")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)